go/chat/push.go

```diff
@@ -655,34 +655,53 @@ func (g *PushHandler) Activity(ctx context.Context, m gregor.OutOfBandMessage) (
 
 			uid := m.UID().Bytes()
 
-			// We need to get this conversation and then localize it
+			// If the topic type is not CHAT or KBFSFILEEDIT, skip the
+			// localization step
 			var inbox types.Inbox
-			if inbox, _, err = g.G().InboxSource.Read(ctx, uid, types.ConversationLocalizerBlocking,
-				types.InboxSourceDataSourceRemoteOnly,
-				nil, &chat1.GetInboxLocalQuery{
-					ConvIDs:      []chat1.ConversationID{nm.ConvID},
-					MemberStatus: chat1.AllConversationMemberStatuses(),
-				}, nil); err != nil {
-				g.Debug(ctx, "chat activity: unable to read conversation: %v", err)
-				return
+			switch nm.TopicType {
+			case chat1.TopicType_CHAT, chat1.TopicType_KBFSFILEEDIT:
+				if inbox, _, err = g.G().InboxSource.Read(ctx, uid, types.ConversationLocalizerBlocking,
+					types.InboxSourceDataSourceRemoteOnly,
+					nil, &chat1.GetInboxLocalQuery{
+						ConvIDs:      []chat1.ConversationID{nm.ConvID},
+						MemberStatus: chat1.AllConversationMemberStatuses(),
+					}, nil); err != nil {
+					g.Debug(ctx, "chat activity: unable to read conversation: %v", err)
+					return
+				}
+			default:
+				if inbox, err = g.G().InboxSource.ReadUnverified(ctx, uid,
+					types.InboxSourceDataSourceRemoteOnly,
+					&chat1.GetInboxQuery{
+						ConvIDs:      []chat1.ConversationID{nm.ConvID},
+						MemberStatus: chat1.AllConversationMemberStatuses(),
+					}, nil); err != nil {
+					g.Debug(ctx, "chat activity: unable to read unverified conversation: %v", err)
+					return
+				}
 			}
-			if len(inbox.Convs) != 1 {
+
+			if len(inbox.ConvsUnverified) != 1 {
```

Comment:
Isn't this always going to fail for chat convs?

Reply:
i don't think so or this line would have already broken: https://github.com/keybase/client/pull/16584/files#diff-50b204c7a958bc27261ba02d3f66bd7eR689